pip install / python setup.py install fail to find pybind11

# Expected behaviour

`pip install simgrid` or `pip setup.py [build] install` work out of the box.

# Issue & Step to reproduce 

e.g on a Grid'5000 nodes using a virtualenv

```
frontend) oarsub -I -l "core=1"
node) virtualenv -p python3 venv
venv) source venv/bin/activate
venv) pip install simgrid
```

Output:
```
[...]
    -- Pybind11 version:
    -- SimGrid needs at least v2.4 of pybind11. Disabling the Python bindings.
    -- Python libs not found. Turn pybind11 off.
    CMake Error at CMakeLists.txt:857 (message):
      Please install pybind11-dev to build the Python bindings (or disable that
      option).

```

Running `python setup.py build install` from the source tree leads to the same
```
venv) git clone --depth 1 https://github.com/simgrid/simgrid.git
venv) cd simgrid
venv) python setup.py build install
```

In both cases pybind11 is installed in the venv (it's declared as a dependency in the setup.py)

```
venv) pip list
Package    Version
---------- -------
pip        20.2
pybind11   2.6.0
setuptools 49.2.0
wheel      0.35.1
```

# Notes

It seems that `pybind11` is  exposing this function `get_cmake_dir` which gives (in the above setup):

```
venv) python -c "import pybind11; print(pybind11.get_cmake_dir())"
/tmp/venv/lib/python3.7/site-packages/pybind11/share/cmake/pybind11
```

So my feeling is that it can be used in the setup.py to force the cmake `find_package` lookup in the right place.
```
        import pybind11
        cmake_args = ['-DCMAKE_LIBRARY_OUTPUT_DIRECTORY=' + extdir,
                      '-DPYTHON_EXECUTABLE=' + sys.executable,
                      '-Denable_smpi=OFF',
                      '-Denable_java=OFF',
                      '-Denable_python=ON',
                      '-Dminimal-bindings=ON',
                      '-Dpybind11_DIR=' + pybind11.get_cmake_dir()]
```

--- a/setup.py
+++ b/setup.py
@@ -47,6 +47,7 @@ class CMakeBuild(build_ext):
             self.build_extension(ext)
 
     def build_extension(self, ext):
+        from pybind11 import get_cmake_dir
         extdir = os.path.abspath(os.path.dirname(
             self.get_ext_fullpath(ext.name)))
         cmake_args = ['-DCMAKE_LIBRARY_OUTPUT_DIRECTORY=' + extdir,
@@ -54,7 +55,9 @@ class CMakeBuild(build_ext):
                       '-Denable_smpi=OFF',
                       '-Denable_java=OFF',
                       '-Denable_python=ON',
-                      '-Dminimal-bindings=ON']
+                      '-Dminimal-bindings=ON',
+                      '-Dpybind11_DIR=' + get_cmake_dir()
+                      ]
 
         cfg = 'Debug' if self.debug else 'Release'
         build_args = ['--config', cfg]

